sympy/physics/tests/test_paulialgebra.py

```diff
@@ -51,6 +51,24 @@ def test_evaluate_pauli_product():
         TensorProduct(I*sigma1*sigma2*sigma1*sigma2, 1)
     ) == 1 -I + I*sigma3*tau1*sigma2 - 1 - sigma3 - I*TensorProduct(1,1)
 
+def test_evaluate_pauli_product_ordering():
+    from sympy.physics.paulialgebra import evaluate_pauli_product
+
+    # Simpler case where ordering is already present
+    assert evaluate_pauli_product(
+        (Pauli(1, label = "a") * Pauli(3, label = "b")) +
+        (Pauli(1, label = "a") * Pauli(3, label = "b"))
+    ) == 2 * Pauli(1, label = "a") * Pauli(3, label = "b")
```

Comment:
It would be a bit easier to read if it was more like:
```
a1 = Pauli(1, label = "a")
b3 = Pauli(3, label = "b")
assert evaluate_pauli_product(a1*b3 + a1*b3) == 2*a1*b3
assert evaluate_pauli_product(a1*b3 + b3*a1) == 2*a1*b3
```